src/d3a/models/area/events.py

```diff
@@ -0,0 +1,87 @@
+from d3a.models.area.event_types import EnableAreaEvent, DisableAreaEvent, ConnectAreaEvent, \
+    DisconnectAreaEvent, DisableIntervalAreaEvent, DisconnectIntervalAreaEvent
+
+
+class IndividualEvents:
+    def __init__(self, event_list, trigger_type):
+        self.event_list = event_list
+        self.trigger_type = trigger_type
+        self._active = True
+
+    def tick(self, current_time):
+        event_list = sorted(self.event_list, key=lambda e: e.event_time)
+        past_events = list(filter(lambda e: e.event_time <= current_time.hour, event_list))
+        self._active = len(past_events) == 0 or type(past_events[-1]) == self.trigger_type
```

Comment:
I have problems understanding the logic here. Shouldn't the Event be active if current_time in event_list.event_time ?

Reply:
No, because the event_list.event_time contain distinct events, and not durations. An example for an event_list = [DisconnectEvent(3), ConnectEvent(5), DisconnectEvent(22)]. In this example the event is going to be active for hours 3-5 and 22-24, which cannot be matched that way. A typical algorithm for handling this case, is to sort the events in chronological order and get all the past events. The past event that is closer to the current time is defining whether the event is active or not. If it is not clear we can discuss this offline.

Reply:
Now I get it. Thanks!